main.go

```diff
@@ -1085,6 +1115,10 @@ func (tf *transformer) transformCompile(args []string) ([]string, error) {
 			return nil, err
 		}
 
+		if tf.curPkg.Name == "main" && strings.HasSuffix(patchMainFile, basename) {
```

Comment:
`patchMainFile` is a global, which makes following this logic pretty tricky. please get rid of the global and instead return more information from `reflectMainPrePatch`

Reply:
or, if you prefer to keep state elsewhere that isn't a local variable, use a field on the transformer type - although I think a variable is probably best as the state is only needed within this func

Reply:
I actually need access to this where using the transformer is not an option, so I would rather keep the global.